zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/QuorumCnxManager.java

```diff
@@ -853,12 +852,30 @@ private void resetConnectionThreadCount() {
      */
     public class Listener extends ZooKeeperThread {
```

Comment:
Shouldn't listener extend ZKCriticalThread? I think this might have gotten lost in the rebase process?

Reply:
I revert this change, because master branch has code which covers my error case. Current code calls System.exit() if(!shutdown and last_exception is SocketException).

Reply:
Ok, thanks.